docs-chef-io/content/server/auth.md

```diff
@@ -386,6 +386,61 @@ common causes:
     minutes. This can be fixed by syncing the clock with an Network Time
     Protocol (NTP) server.
 
+## User public key update
+
+Chef infra server user can update public_key using the following methods.
+-   Using reregister method
+    Use the reregister argument in the knife command to regenerate an RSA key pair for a user. The public key will be stored on the Chef Infra Server and the private key will be displayed on STDOUT or written to a named file.
+    
+    Syntax:- knife user reregister USER_NAME (options)
+    Ex:- knife user reregister "Arjun Koch" 
+
+-  Using knife user key command
+    User can have multiple public keys associated with his chef server account.  Each public key has a name and expiration date associated with it.
+    Name is an ASCII string used to identify the public key.  The expiration date for the public key is specified as an ISO 8601 formatted string: YYYY-MM-DDTHH:MM:SSZ.
+    User can list, add, edit and delete public keys with knife user key command.  The following operations supported.
+        knife user key create to create public key.
+        knife user key delete to delete public key.
+        knife user key list to list user public keys.
+        knife user key edit to edit public keys.
+
+    Constraints: User can't edit key details which is currently in use.
+
+    Ex:- To change the key pair of user1.
+       1) Check current keys associated with the user with the help of knife user key list USERNAME.
+          #knife user key list user1
+          default
+
+       2) Create new key pair with the help of knife user key create.
+          #knife user key create user1 --key-name new_key_03_11_2022
+
+              {
+                "user": "user1",
+                "name": "new_key_03_11_2022",
+                "expiration_date": "infinity",
+                "create_key": true
+              }
```

Comment:
Is this generated data from knife? Does the user need to do something with it?

Reply:
This is knife generated content {generated with arguments passed to the knife command} which appears in a text editor where user can do modifications if required.

Reply:
Ok, does knife generate a file and place it somewhere when you run that command? Does this text already exist and the user has to update it, or does knife overwrite an existing file? I assume knife automatically opens the text editor, correct?